### PR TITLE
Validate schemas overlooking the nullability fields

### PR DESCRIPTION
**Pull request by ireneisdoomed**
This PR includes:
- [x] Changes to the validate_schema Dataset method to compare schemas between dataframes by looking at the names.
    - The functionality is essentially the same, only that we iterate on the column names, instead of on the properties of each schema field.
    - There are 2 checks in place: a) checking that the observed schema does not have any extra field; b) checking that the mandatory fields dictated by the expected schema are present in the observed_schema
- [x] Added 2 unit tests for each of the functionalities I describe above

Notes:
- I also had to tweak the `pytest_generate_tests` function. TIL that when we use pytest's metafuncs, we are defining a fixture that will be used in every test, unless specified otherwise.
- Mypy doesn't check the script that tests the schema. This is because mypy is crashing when checking this script, even with the newer version, hence I was not able to commit. It's not that the script is raising a typing error, but mypy is failing to deserialize something. I've found an [issue](https://github.com/python/mypy/issues/12257) in their repo that might be related. Full trace:
```
Traceback (most recent call last):
  File "/Users/irenelopez/.cache/pre-commit/repomwnp37wt/py_env-python3.8/bin/mypy", line 8, in <module>
    sys.exit(console_entry())
  File "/Users/irenelopez/.cache/pre-commit/repomwnp37wt/py_env-python3.8/lib/python3.8/site-packages/mypy/__main__.py", line 15, in console_entry
    main()
  File "mypy/main.py", line 95, in main
  File "mypy/main.py", line 174, in run_build
  File "mypy/build.py", line 193, in build
  File "mypy/build.py", line 276, in _build
  File "mypy/build.py", line 2903, in dispatch
  File "mypy/build.py", line 3284, in process_graph
  File "mypy/build.py", line 3362, in process_fresh_modules
  File "mypy/build.py", line 2101, in load_tree
  File "mypy/nodes.py", line 397, in deserialize
  File "mypy/nodes.py", line 3689, in deserialize
  File "mypy/nodes.py", line 3630, in deserialize
  File "mypy/nodes.py", line 274, in deserialize
  File "mypy/nodes.py", line 854, in deserialize
  File "mypy/types.py", line 189, in deserialize_type

  File "mypy/types.py", line 2038, in deserialize
KeyError: 'unpack_kwargs'
```

### CODE REVIEW

**Review by d0choa — APPROVED** [review]
Do you think is worth considering the next 2 cases?

- use `StructField` `.dataType` method to compare if the expected and observed types are the same. I believe they are ignored at the moment. (e.g. `df.schema[0].dataType`)
- While we were comparing the whole `StructField` before, this PR only compares the names. This makes all nested information ignored. If I understand this correctly you would be ignoring all differences in nested columns. We might need some recursivity similar to this [chunk of code from Chispa](https://github.com/MrPowers/chispa/blob/e1ea9b714c1385baa522f34a050421eb7bc766b5/chispa/schema_comparer.py#L67).

Since you have some tests already, perhaps it's useful to add some of these cases and try to do test-driven development.

We can also merge and work separately if this unblocks other work. I'm approving just in case.

**Comment by DSuveges** [comment]
I agree with @d0choa, we should only drop the nullability check not the type check.

**Comment by ireneisdoomed** [comment]
@d0choa @d0choa Thank you so much for your comments. They were all on point. The new changes include:
- types of each field are now validated, not only the names (leaving apart only the nullability property)
- compatibility for nested validation. this is achieved by flattening the schema of both dataframes to extract their fields. the function is heavily inspired in the one we have to calculate the metrics.
- unit tests for: missing field, extra field, field of different type. run both in nested and not nested mock data.

File: .gitignore
Comment (d0choa) [inline .gitignore:1]: I don't think this belongs here

**Comment by ireneisdoomed** [comment]
Last change: added an ad hoc check for duplicated fields in https://github.com/opentargets/genetics_etl_python/pull/71/commits/d0b34891e16bfa48aaf2c85170d562d8a70fe47c

**Pull request merged**